main.go

```diff
@@ -0,0 +1,205 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+	"sort"
+	"strings"
+
+	"github.com/bitrise-community/steps-ionic-prepare/cordova"
+	"github.com/bitrise-community/steps-ionic-prepare/ionic"
+	"github.com/bitrise-community/steps-ionic-prepare/npm"
+	"github.com/bitrise-io/go-utils/colorstring"
+	"github.com/bitrise-io/go-utils/command"
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-io/go-utils/pathutil"
+	"github.com/bitrise-tools/go-steputils/stepconf"
+	ver "github.com/hashicorp/go-version"
+)
+
+type config struct {
+	Platform              string `env:"platform,opt[ios,android,'ios,android']"`
+	Readd                 bool   `env:"readd_platform,opt[true,false]"`
+	IonicVersion          string `env:"ionic_version"`
+	CordovaVersion        string `env:"cordova_version"`
+	CordovaIosVersion     string `env:"cordova_ios_version"`
+	CordovaAndroidVersion string `env:"cordova_android_version"`
+	WorkDir               string `env:"workdir,dir"`
+}
+
+func getField(c config, field string) string {
```

Comment:
Please strict the usability with a receiver for a function if you do a type independent logic but with a type given in an argument. Also please do not use `Indirect` if not needed. So this method is only for your config type, so please move the method under it's hierarchy. Please keep in mind that this logic will not throw any errors but will return a string `<invalid Value>` in case if you call it with a field name that doesn't exists.

```
func(c config) getField(field string) string {
	return reflect.ValueOf(c).FieldByName(field).String()
}
```

otherwise 😍 and a very nice dynamic use case